org.bardsoftware.eclipsito/src/org/bardsoftware/impl/eclipsito/BootImpl.java

```diff
@@ -35,6 +39,42 @@ public void run(String application, File modulesDir, String descriptorPattern, L
       return plugins;
     }
 
+  public File getVersionDir(List<File> modulesDir) {
+    String version = null;
+    String pluginsPath = null;
+    if (modulesDir != null) {
```

Comment:
Just add precondition: `assert modulesDir != null` in the very beginning of this method. modulesDir being null means programmer error, it is not an expected case.